lib/babe/verify.go

```diff
@@ -327,62 +327,58 @@ func (b *verifier) verifyAuthorshipRight(header *types.Header) error {
 		return ErrBadSignature
 	}
 
-	// check if the producer has equivocated, ie. have they produced a conflicting block?
-	// hashes is hashes of all blocks with same block number as header.Number
-	hashes := b.blockState.GetAllBlocksAtDepth(header.ParentHash)
+	equivocated, err := b.verifyBlockEquivocation(header)
+	if err != nil {
+		return fmt.Errorf("could not verify block equivocation for header %s: %w", header.Hash(), err)
```

Comment:
header should be part of the error context set by verifyBlockEquivocation, not the caller (to avoid repeating it):

```suggestion
		return fmt.Errorf("could not verify block equivocation: %w", err)
```

Reply:
> to avoid repeating it

desc gets printed only once in this case. I would rather add that context here once than adding it in all errors in `verifyBlockEquivocation`

Reply:
Problem is we should standardize this; you could have a local function within `verifyBlockEquivocation`:

```go
func verifyBlockEquivocation(...) {
   wrapError := func(err error) error {
     return fmt.Errorf("for block header %s: %w", header, err)
  }
  // ...
  if err != nil {
    return false, wrapError(err)
  }

}
```

Reply:
No since it opens the door to adding multiple times the same context at different levels of the call stack.
Adding the context once at the bottom layer actually handling the argument is the way to avoid this and not have to think about it.